docs/web-apps/automated-testing/pre-run-executables.md

```diff
@@ -0,0 +1,308 @@
+sudo access is not allowed on Linux/Mac virtual machines. Administrative access is not allowed on Windows VMs.
+
+### Writing a Configuration Script
+This example script disables the warning that Safari pops up when using basic HTTP authentication to access a website.
+
+**disable_fraud.sh**
+```
+#!/bin/bash
+defaults write com.apple.Safari WarnAboutFraudulentWebsites false
+```
+
+>**NOTE:** Match File Types to Your Operating System
+When creating your executable file, take into account the operating system you'll use for your tests. For example, Bash commands won't work on Windows machines, which instead require a `.bat` or `.com` file.
+
+### Storing a Configuration Script
+Your script can be stored in GitHub or in [Sauce Storage](https://wiki.saucelabs.com/display/DOCS/Temporary+Storage+Methods). You can also use [Gist](https://gist.github.com/) to easily host your executable file. Make sure to use the link containing the raw file contents.
+
+###Set the `prerun` Capability
+This example sets `prerun` to point to `myscriptstorage.com`, which hosts the script `disable_fraud.sh` used as an example in step 1.
+
+```
+desired_capabilities['prerun'] = {
+    'executable':'http://myscriptstorage.com/disable_fraud.sh',
+    'background': 'false'
+}
+```
+
+This example accesses the same script from Sauce Storage:
+
+```
+desired_capabilities['prerun'] = {
+    'executable':'sauce-storage:disable_fraud.sh',
+    'background': 'false'
+}
+```
+
+## Setting Up Silent Mode
+One of the arguments that can be added to the `prerun` capability in your test configuration options is `--silent`, which can be abbreviated to `/S`. This argument will "silently" install the executable without any modal dialogs or other installation processes showing up in your tests. Here's an example of how you would set the prerun capability to run in silent mode:
+
+```
+"prerun": {
+    "executable": "http://url.to/your/executable.exe",
+    "args": [ "/S", "-a", "-q" ],
+    "background": true,
+    "timeout": 120
+}
+```
+
+## Setting Basic Authentication in Safari
+In Safari, when you try to navigate to a URL that uses basic HTTP authentication, a warning dialog pops up that will derail your automated tests unless you can resolve it. You can use a short shell script that you execute as a pre-run executable to configure Safari to not show this message.
+
+>**NOTE:** You can only use this for Safari versions up to and including 10.x.
+
+This script clears the Warn when visiting a fraudulent website option in Safari's preferences, so that your automated tests that use basic authentication over HTTP will execute without triggering the warning.
+
+```
+#!/bin/bash
+defaults write com.apple.Safari WarnAboutFraudulentWebsites false
+```
+
+[This script](https://gist.githubusercontent.com/saucyallison/3a73a4e0736e556c990d/raw/d26b0195d48b404628fc12342cb97f1fc5ff58ec/disable_fraud.sh) is hosted as a Gist on GitHub.
+
+### Setting the `prerun` Capability
```

Comment:
This section seems like mostly a duplicate of the previous `### Set the `prerun` Capability` under the `## Writing a Configuration Script` section? Is there any value to having this same level of info in here twice? Maybe we should bubble the topic up to a H2 level and present it on its own?